mod/photos.php

```diff
@@ -654,7 +654,11 @@ function photos_post(App $a)
 
 	Logger::info('photos: upload: received file: ' . $filename . ' as ' . $src . ' ('. $type . ') ' . $filesize . ' bytes');
 
-	$maximagesize = DI::config()->get('system', 'maximagesize');
+	$maximagesize = Strings::getBytesFromShorthand(DI::config()->get('system', 'maximagesize'));
+
+	if ($maximagesize == 0) {
+		$maximagesize = INF;
+	}
 
 	if ($maximagesize && ($filesize > $maximagesize)) {
```

Comment:
This condition was already effectively using `0` to disable the check, not sure using `INF` here adds anything.

Reply:
That is the point, if we pass `maximagesize == 0` to line 663, the processing will be discarded and since `fileesize` will always `> 0`. And this is contradicting the semantic that `maximagesize == 0` should effect "no limit"

Reply:
Ahh sorry, now I get it (as you already guessed I am not too familiar with PHP ;-) ) since the AND condition is already false because PHP trades "0" as false. I get it. You are right, then it is not necessary.

Reply:
There are two variants of this if statement in line 663, the other one is `!isempty($maxfilesize)`. I will adjust it by removing `isempty()` .

Reply:
Thanks, it bothered me too but I didn't want to pile on you.